pkg/resourcesets/collector.go

```diff
@@ -249,156 +229,79 @@ func (h *ResourceHandler) filterByLabel(ctx context.Context, dr dynamic.Resource
 }
 
 func (h *ResourceHandler) filterByName(filter v1.ResourceSelector, resourceObjectsList []unstructured.Unstructured) ([]unstructured.Unstructured, error) {
-	var filteredByName, filteredByResourceNames []unstructured.Unstructured
+	var filteredByName []unstructured.Unstructured
 
 	if len(filter.ResourceNames) == 0 && filter.ResourceNameRegexp == "" && filter.ExcludeResourceNameRegexp == "" {
 		// no filters for names of the resource, return all objects obtained from the list call
 		return resourceObjectsList, nil
 	}
 
-	filteredByNameMap := make(map[string]bool)
+	// Map to store included names for fast lookup
+	includeNameMap := make(map[string]bool)
+	for _, name := range filter.ResourceNames {
+		includeNameMap[name] = true
+	}
 
-	// filter out using ResourceNameRegexp
+	var includeRegex *regexp.Regexp
 	if filter.ResourceNameRegexp != "" {
-		logrus.Debugf("Using ResourceNameRegexp [%s] to filter resource names", filter.ResourceNameRegexp)
-		for _, resObj := range resourceObjectsList {
-			if filter.ResourceNameRegexp != "." {
-				metadata := resObj.Object["metadata"].(map[string]interface{})
-				name := metadata["name"].(string)
-				nameMatched, err := regexp.MatchString(filter.ResourceNameRegexp, name)
-				if err != nil {
-					return filteredByName, err
-				}
-				if !nameMatched {
-					logrus.Debugf("Skipping [%s] because it did not match ResourceNameRegexp [%s]", name, filter.ResourceNameRegexp)
-					continue
-				}
-			}
-			filteredByName = append(filteredByName, resObj)
-			key := resObj.Object["filterUid"].(string)
-			filteredByNameMap[key] = true
-		}
+		includeRegex = regexp.MustCompile(filter.ResourceNameRegexp)
```

Comment:
I don't know if we have a pre-processing pass elsewhere in the code that validates the user input filters before it reaches this function, otherwise we probably want to avoid panicking if the regex is invalid.